Sources/StreamVideo/Controllers/CallController.swift

```diff
@@ -80,9 +80,10 @@ class CallController: @unchecked Sendable {
         Task {
             await handleParticipantCountUpdated()
             let participantsPublisher = await webRTCCoordinator.stateAdapter.$participants
-            self.debouncedParticipants = CollectionDebouncedUpdateObserver(
+            self.participants = CollectionDelayedUpdateObserver(
                 publisher: participantsPublisher.eraseToAnyPublisher(),
-                initial: [:]
+                initial: [:],
+                mode: .throttle(scheduler: DispatchQueue.main, latest: true)
```

Comment:
This is where the participants updates are getting throttled to not overwhelm the UI rendering. By default the `CollectionDelayedUpdateObserver` is using the screen refresh rate as the default delay. As more participants joinining the delay increases to improve performance during handling big volume of events.